sdk/tests/conformance/extensions/webgl-webcodecs-video-frame.html

```diff
@@ -0,0 +1,220 @@
+
+      const init = {
+        output: (chunk) => {
+          testPassed("Encode frame successfully.");
+          pending_outputs--;
+          processChunk(chunk);
+        },
+        error: (e) => {
+          testFailed("Failed to encode frame.");
+          finishTest();
+          vtr.stop();
+        }
+      };
+
+      const config = {
+        codec: codec_string,
+        width: cnv.width,
+        height: cnv.height,
+        bitrate: 10e6,
+        framerate: fps,
+      };
+
+      let encoder = new VideoEncoder(init);
+      encoder.configure(config);
+
+      vtr.start((frame) => {
+        if (pending_outputs > 30) {
+          console.log("drop this frame");
+          // Too many frames in flight, encoder is overwhelmed
+          // let's drop this frame.
+          return;
+        }
+        if (frame_counter == kMaxFrame) {
+          return;
+        }
+
+        frame_counter++;
+        pending_outputs++;
+        const insert_keyframe = (frame_counter % 150) == 0;
+        encoder.encode(frame, { keyFrame: insert_keyframe });
+      });
+    }
+
+    function startDecodingAndRendering(cnv, handleFrame) {
+      const init = {
+        output: handleFrame,
+        error: (e) => {
+          testFailed("Failed to decode frame.");
+          finishTest();
+        }
+      };
+
+      const config = {
+        codec: codec_string,
+        codedWidth: cnv.width,
+        codedHeight: cnv.height,
+        acceleration: "deny",
+
+      };
+
+      let decoder = new VideoDecoder(init);
+      decoder.configure(config);
+      return decoder;
+    }
+
+    // Compare pixels in 4 corner (upper left, upper right, bottom left, bottem right) and center
+    // with given color to decide whether the frame is matched.
+    function isFramePixelMatched(gl, th_per_pixel = pixelCompareTolerance) {
+      let pixels = new Uint8Array(src_width * src_height * 4);
+      gl.readPixels(0, 0, src_width, src_height, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
+      const kPixelPositions = [0, //upper left
+        (src_width - 1) * 4, //upper right
+        (src_width * Math.floor(src_height / 2) + Math.floor(src_width / 2)) * 4, // center
+        (src_width * (src_height - 1) - 1) * 4, //bottom left
+        (src_width * src_height - 1) * 4]; // bottem right
+      let delta_all = 0;
+      for (let i = 0; i < kPixelPositions.length; i++) {
+        let delta = Math.abs(pixels[kPixelPositions[i]] - kTestPixel[0]) // R
+          + Math.abs(pixels[kPixelPositions[i] + 1] - kTestPixel[1]) // G
+          + Math.abs(pixels[kPixelPositions[i] + 2] - kTestPixel[2]) // B
+        delta_all += delta;
+
+      }
+      delta_all = delta_all / kPixelPositions.length;
+      console.log("Average pixel difference:" + delta_all);
+      if (delta_all > th_per_pixel)
+        return false;
+      return true;
+    }
+
+    function main() {
+      if (!("VideoEncoder" in window)) {
+        testFailed("WebCodecs API is not supported.");

```

Comment:
The tests in conformance/extensions need to skip themselves if the extension isn't supported - which is always legal. Could you please update this to be similar to some of the other recently-added tests in this directory?

Reply:
I fixed code here. How about create a new PR to deal with related cases in other files?